packages/node/test/integrations/http.test.ts

```diff
@@ -197,6 +197,39 @@ describe('tracing', () => {
     expect(loggerLogSpy).toBeCalledWith('HTTP Integration is skipped because of instrumenter configuration.');
   });
 
+  it('omits query and fragment from description and adds to span data instead', () => {
+    nock('http://dogs.are.great').get('/spaniel?tail=wag&cute=true#learn-more').reply(200);
+
+    const transaction = createTransactionOnScope();
+    const spans = (transaction as unknown as Span).spanRecorder?.spans as Span[];
+
+    http.get('http://dogs.are.great/spaniel?tail=wag&cute=true#learn-more');
+
+    expect(spans.length).toEqual(2);
+
+    // our span is at index 1 because the transaction itself is at index 0
+    expect(spans[1].description).toEqual('GET http://dogs.are.great/spaniel');
+    expect(spans[1].op).toEqual('http.client');
+    expect(spans[1].data.method).toEqual('GET');
+    expect(spans[1].data.url).toEqual('http://dogs.are.great/spaniel');
+    expect(spans[1].data['http.query']).toEqual('tail=wag&cute=true');
+    expect(spans[1].data['http.fragment']).toEqual('learn-more');
+  });
+
+  it('omits username and password entirely from span', () => {
+    nock('http://username:password@dogs.are.great').get('/').reply(200);
+
+    const transaction = createTransactionOnScope();
+    const spans = (transaction as unknown as Span).spanRecorder?.spans as Span[];
+
+    http.get('http://username:password@dogs.are.great/');
+
+    expect(spans.length).toEqual(2);
+
+    // our span is at index 1 because the transaction itself is at index 0
+    expect(spans[1].description).toEqual('GET http://dogs.are.great/');
```

Comment:
It would be better if the resulting URL is: `http://[Filtered]:[Filtered]@dogs.are.great/` so it is clear to the user that there once was a username/pwd in the URL and it was now removed.

Reply:
Ok, I'll make that change.

What are your thoughts on https://github.com/getsentry/sentry-javascript/pull/7206#discussion_r1108702894 ?

Noting that `[Filtered]:[Filtered]@` is similar breaking change. Today if I configure:
```ts
shouldCreateSpanForRequest: (url) => url.startsWith('https://example.com')
```
And then I request `https://user:pass@example.com`, the span will be recorded, but with `[Filtered]:[Filtered]` it would not be recorded